File: Catalogo/Parte2/FiltroDominioFrecuencia/main.m
clc; clear; close all;
pkg load image;

A = imread("edificio_china.jpg");

Y_ilpf = ideal_lpf(A,100);
figure(1);
subplot(1,2,1);
imshow(A);
title("Original image");
subplot(1,2,2);
imshow(Y_ilpf);
title("Ideal (LPF)");

Y_blpf = butterworth_lpf(A,100,2);
figure(2);
subplot(1,2,1);
imshow(A);
title("Original image");
subplot(1,2,2);
imshow(Y_blpf);
title("Butterworth (LPF)");

Y_ghpf = gaussian_hpf(A,2.0);
figure(3);
subplot(1,2,1);
imshow(A);
title("Original image");
subplot(1,2,2);
imshow(Y_ilpf);
title("Gaussian (HPF)");

File: Catalogo/Parte2/FiltroDominioFrecuencia/main.py
from utils import *
from dft2d import *
from gaussian_lpf import *
from ideal_hpf import *
from butterworth_hpf import *

A = read_image("chest.jpg")
X = read_image("edificio_china.jpg")

print("Calculating ...")
F = dft2d(A)
print("Done ...")
plot_image(np.abs(F))

Y_glpf = gaussian_lpf(X=X, sigma=2.0)
compare_images(X, Y_glpf, "Original Image", "Gaussian (LPF)")

Y_ihpf = ideal_hpf(X=X, D0=100)
compare_images(X, Y_ihpf, "Original Image", "Ideal (HPF)")

Y_bhpf = butterworth_hpf(X=X, D0=100, n=2)
compare_images(X, Y_bhpf, "Original Image", "Butterworth (HPF)")
